src/imageio/format/jpeg.c

```diff
@@ -396,6 +394,10 @@ int write_image(dt_imageio_module_data_t *jpg_tmp, const char *filename, const v
   jpeg_finish_compress(&(jpg->cinfo));
   jpeg_destroy_compress(&(jpg->cinfo));
   fclose(f);
+
+  if(exif && exif_len > 0 && exif_len < 65534)
```

Comment:
(http://dev.exiv2.org/issues/1221#note-1 happens with the old, jpeg-native way of writing metadata, i have already checked that)